wandb/sdk/lib/hashutil.py

```diff
@@ -48,15 +47,25 @@ def md5_file_hex(*paths: StrPath) -> HexMD5:
     return HexMD5(_md5_file_hasher(*paths).hexdigest())
 
 
+_MIN_CHUNKED_FILESIZE: int = 1_024 * 1_024
+"""Files larger than this size (bytes) should be read in chunks to conserve memory."""
+
+_CHUNKSIZE: int = _MIN_CHUNKED_FILESIZE // 128
```

Comment:
It might be worth doing some benchmarking; in tests I ran (probably more than a year ago at this point) both my laptop and a few *nix cloud VMs operated significantly better with 1-16MB chunksizes than the 4kB  chunksizes that you see in example code everywhere, on both spinning magnetic and SSD HDs.

Reply:
Updated comment after looking at some of my early benchmarks from https://gist.github.com/moredatarequired/01f8c22907e33c130955235337c875ba. 128kB+ was actually fine, the ideal being about 512kB, being about 3x faster than 4kB and 50% faster than 64kB (on my machine specifically, but  I saw the same conclusions with smaller differences on other machines and configurations)

Reply:
Good point and yeah, gut feeling is that larger chunk sizes (well, larger than 4kB) are going to be faster here.

I can do some adhoc benchmarking, but to simplify things, maybe we can just use the same variable `_CHUNKSIZE` for both 
- the chunk size 
- the filesize threshold
(and use 1MB as we already do for the threshold)

Reply:
whoops, missed your 2nd comment -- this is useful, thanks!  Looking like it might better to keep _CHUNKSIZE separate from the threshold after all

Reply:
Honestly, if it's only a fallback for a corner case, I don't care that much, and we'd want to tune to mounted file systems anyway. I would use 128kB and do an unconditional chunked read (drop the whole file read).

Reply:
revamped the implementation a bit, including dropping the whole file read like you suggested (mmap should handle this anyway, save for empty files which are skipped)

Reply:
Also full disclosure, I may have cleaned up the test module (`test_hashutil.py`) a bit too.  Nothing major though